### DISCUSSION TITLE
OpenGWAS ontology mapping

### DISCUSSION THREAD

**Discussion by explodecomputer** [MRCIEU/opengwas#7] [General]
# OpenGWAS Ontology mapping project

Ontology mapping can be approached with different attitudes. The approach we would like to take for OpenGWAS is one in which there is a strict correspondence between the reported trait, and the mapped ontology. In some instances the reported trait will not have a strict ontology available. 

Each trait has been assigned three potential ontology terms through an automated mapping system. The automated mapping system is not perfect, and we need to manually validate each of these terms.

## Instructions

For each row in the spreadsheet:

1. Some traits are too ambiguous to be assigned to an EFO term. See info below about how to identify such traits. If this trait is 'ambiguous' then it needs to be assigned a 'broad' category (rather than a strictly corresponding EFO term). In the Broad/Narrow column, mark the trait as 'Broad'. Otherwise, mark the trait as 'Narrow'.
2. Determine if the reported trait matches any of the three automaticilly mapped ontologies. If it does, assign 1/2/3 in the 'Check' column
3. If none of the automatically mapped ontologies are appropriate then search on https://www.ebi.ac.uk/efo/ for a more appropriate trait. 
4. If an appropriate mapping is not available within EFO, and the trait seems sufficiently well characterised that it should have an EFO term, mark the 'Check' column with 'EFO needed'
5. In the comments section provide any other notes e.g. issues with the trait name

## Ambiguous traits

- Treatment, medications or operations seldom have ontologies and are difficult to map. These can be assigned to a broad category of 'treatment' EFO:0000727
- Occupation questions are broad - use a broad category of 'employment status' EFO:0005241
- Any food intake questions can be assigned to a broad category of 'diet measurement' EFO:0008111
- ICD10 codes can be ignored (will use the actual ICD10)
- Operation procedures can be ignored (will use the actual OPCS)
- Sometimes traits are nested e.g. 'Why stopped smoking: illness or ill health' is a follow up question to smoking status, and this can be assigned a broad category e.g. 'smoking behaviour' EFO:0004318
- Other examples of nested trait e.g. 'Stronger worrying (than other people) during period of worst anxiety' - level of worry during anxiety. It's too complex to be captured by an ontology as it relates to multiple concepts, so use a broad one.
- There may be other things like 'Time spent watching television', that are slightly ambiguous in what they're actually measuring. You could assign this to 'physical activity' for example

**Comment by explodecomputer** [comment]
@rsgoncalves has provided this updated automated mapping https://docs.google.com/spreadsheets/d/1XJK-DmoJZrMKBPABBdoqWcxjvm4Qix0QVrQl4K6aFbA.

**Comment by explodecomputer** [comment]
Update meeting with Phil, Tom, Gib 2022-02-11

## Order of activities:

1. Meeting with everyone to go through the task
  - Introduce the problem and what we're trying to achieve
  - Introduction to ontologies
  - Use 'body mass index' as an easy example
  - Use 'blood sugar' as a harder example for discussion
2. Provide everyone with 100 traits to map. Some overlap between them.
3. Everyone provides results for their 100 traits.
4. Meeting with everyone to discuss and talk through the mappings, potentially update documentation or methodology
5. Provide everyone with the rest of the traits to map. Some overlap between them.
6. Everyone provides results for the remaining traits

This is the latest mapping: https://docs.google.com/spreadsheets/d/1Gu6TKvi4u4yKXeR5p2DbpTpa5urgIp3VsAxGXH3hNKI/edit#gid=1905015624

We add these columns:

- Exact 
  - Trait and mapped-to term are deemed fully equivalent to one another in definition
  - Primarily choose this column
  - EFO only
  - One term only
- Broad
  - Mapped-to term is a larger concept than the trait itself. It may be an umbrella concept covering the query trait
  - EFO only
  - Multiple terms allowed
  - Only if no exact EFO
- Narrow
  - Mapped-to term is a smaller concept than the trait itself. It may be that the query trait is an umbrella concept covering the mapped-to term
  - EFO only
  - Multiple terms allowed
  - Only if no exact EFO
- Inadequate
  - A related term that may be useful but doesn't strictly fall under broad or narrow
  - EFO only
  - Only if no exact, broad or narrow EFO terms
- Exact non-EFO
  - if there is no exact EFO mapping then find an exact non-EFO mapping, look in this order:
    - SNOMED
    - ICD10
    - HPO
    - MESH
- Comments
  - Provide info for further action, e.g. some options:
    - Request new EFO term
    - Consult with EBI
    - Issue with the trait name
    - Other

**Reply by mightyphil2000** [reply]
**EFOs:**
https://www.ebi.ac.uk/efo/

**Links to non-EFO resources:**
SNOMED - https://bioportal.bioontology.org/
ICD10 - https://www.icd10data.com/
HPO - https://hpo.jax.org/app/
MESH - https://www.ncbi.nlm.nih.gov/mesh/

**Comment by rsgoncalves** [comment]
In this document is a brief intro to ontologies and EFO: https://docs.google.com/document/d/1sPzTmWEzJkfZz3TVI4EgHTgiY6M7u_nU. Planning to add concrete examples related to the mapping exercise, and later on observations and directions based on the pilot.
